### DISCUSSION TITLE
mitsuba 3 is offline ray tracing renderer?

### DISCUSSION THREAD

**Discussion by bipul-mohanto** [mitsuba-renderer/mitsuba3#784] [Q&A] [answered]
I am searching for a real-time path-tracing rendering engine for research purposes, where at least the naive path tracer is built-in.

I know mitsuba 2 and 3 are very popular in the physically-based rendering community. I checked some documentation of mitsuba 3. My initial guesses are:

* mitsuba has only ray tracing algorithm in it, not path tracing. Am I right?
* The ray tracing is offline, not real-time. Am I right? 

Please correct me if I am wrong.

**Comment by njroussel** [comment, marked as answer]
Hi @bipul-mohanto 

Mitsuba has a standard path tracer implementation available ([documentation](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#path-tracer-path)). If you're looking for an algorithm where the rays start at a light source, this is also possible with Mitsuba using the [`ptracer`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#particle-tracer-ptracer).

Mitsuba is not a real-time renderer. There is an important focus on performance, but real-time performance is not an immediate goal of the project.

**Reply by bipul-mohanto** [reply]
Thanks @njroussel for your clarification.
